cli/azd/pkg/ext/hooks_manager.go

```diff
@@ -0,0 +1,204 @@
+	validHookNames := []string{}
+
+	for _, commandName := range commands {
+		// Convert things like `azd infra create` => 'configlist`
+		commandName = strings.TrimPrefix(commandName, "azd")
+		commandName = strings.TrimSpace(commandName)
+		commandName = strings.ReplaceAll(commandName, " ", "")
+
+		validHookNames = append(validHookNames, strings.ToLower(string(prefix)+commandName))
+	}
+
+	predicate := func(scriptName string, scriptConfig *ScriptConfig) bool {
+		index := slices.IndexFunc(validHookNames, func(hookName string) bool {
+			return hookName == scriptName
+		})
+
+		return index > -1
+	}
+
+	return h.filterScriptConfigs(scripts, predicate)
+}
+
+// Filters the specified script configurations based on the predicate
+// Will return an error if any configuration errors are found
+func (h *HooksManager) filterScriptConfigs(
+	scripts map[string]*ScriptConfig,
+	predicate HookFilterPredicateFn,
+) ([]*ScriptConfig, error) {
+	allHooks := []*ScriptConfig{}
+	explicitHooks, err := h.getExplicitHooks(scripts, predicate)
+	if err != nil {
+		return nil, err
+	}
+
+	implicitHooks, err := h.getImplicitHooks(predicate)
+	if err != nil {
+		return nil, err
+	}
+
+	allHooks = append(allHooks, explicitHooks...)
+
+	// Only append implicit hooks that were not already wired up explicitly.
+	for _, implicitHook := range implicitHooks {
+		index := slices.IndexFunc(allHooks, func(hook *ScriptConfig) bool {
+			return implicitHook.Name == hook.Name
+		})
+
+		if index >= 0 {
+			log.Printf(
+				"Skipping hook @ '%s'. An explicit hook for '%s' was already defined in azure.yaml.\n",
+				implicitHook.Path,
+				implicitHook.Name,
+			)
+			continue
+		}
+
+		allHooks = append(allHooks, implicitHook)
+	}
+
+	return allHooks, nil
+}
+
+func (h *HooksManager) getExplicitHooks(
+	scripts map[string]*ScriptConfig,
+	predicate HookFilterPredicateFn,
+) ([]*ScriptConfig, error) {
+	matchingScripts := []*ScriptConfig{}
+
+	// Find explicitly configured hooks from azure.yaml
+	for scriptName, scriptConfig := range scripts {
+		if scriptConfig == nil {
+			continue
+		}
+
+		if predicate != nil && !predicate(scriptName, scriptConfig) {
+			continue
+		}
+
+		// If the script config includes an OS specific configuration use that instead
+		if runtime.GOOS == "windows" && scriptConfig.Windows != nil {
+			scriptConfig = scriptConfig.Windows
+		} else if (runtime.GOOS == "linux" || runtime.GOOS == "darwin") && scriptConfig.Posix != nil {
+			scriptConfig = scriptConfig.Posix
+		}
+
+		scriptConfig.Name = scriptName
+		scriptConfig.Path = strings.ReplaceAll(scriptConfig.Path, "/", string(os.PathSeparator))
+
+		if err := scriptConfig.validate(); err != nil {
+			return nil, fmt.Errorf("hook configuration for '%s' is invalid, %w", scriptName, err)
+		}
+
+		matchingScripts = append(matchingScripts, scriptConfig)
+	}
+
+	return matchingScripts, nil
+}
+
+func (h *HooksManager) getImplicitHooks(predicate HookFilterPredicateFn) ([]*ScriptConfig, error) {
```

Comment:
How bad are things if we remove the idea of implicit hooks for now?  I'm sort of worried that this is going to push folks towards trying to check in portions of `.azure` which I think is something we really don't want.

I'm also slightly nervous about this directory being interpreted as an environment at some level (since right now we don't segment all the environment under a `envs` folder or something under `.azure`.